ifrs17-template/Test/ReimportTest.ipynb

```diff
@@ -0,0 +1,291 @@
+{
+    "metadata": {
+        "authors": [],
+        "id": "VOS67A7-B0Omk1s4TFC3LQ",
+        "kernelspec": {
+            "display_name": "Formula Framework",
+            "language": "C#",
+            "name": "C#"
+        },
+        "language_info": {
+            "file_extension": ".cs",
+            "mimetype": "text/plain",
+            "name": "C#"
+        }
+    },
+    "nbformat": 4,
+    "nbformat_minor": 5,
+    "cells": [
+        {
+            "cell_type": "markdown",
+            "source": [
+                "<p style=\"font-weight:bold;\"> <span style=\"font-size: 36px\"> Reimport of Transactional Data Test</span> </p>"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "#!eval-notebook \"../Initialization/InitSystemorphBaseToMemory\""
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Check Method"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task<ActivityLog> ReimportAndCompare((string transactionalData, string importFormat) importData, DataNodeState[] dataNodeStates)",
+                "\n{",
+                "\n    var workspace = Workspace.CreateNew();",
+                "\n    workspace.Initialize(x => x.FromSource(DataSource).DisableInitialization<DataNodeState>());",
+                "\n    await workspace.UpdateAsync<DataNodeState>(dataNodeStates);",
+                "\n",
+                "\n    Activity.Start();",
+                "\n    var importActivity = await Import.FromString(importData.transactionalData).WithFormat(importData.importFormat).WithTarget(workspace).ExecuteAsync();",
+                "\n    if(importActivity.RepeatOnce().ToArray().HasErrors()) return importActivity;",
```

Comment:
I think you don't need .RepeatOnce().ToArray()

Reply:
It is in fact necessary, without it I get the following error:

<img width="986" alt="image" src="https://github.com/Systemorph/IFRS17CalculationEngine/assets/114566025/059d253a-78ac-41c7-955f-748d31687484">